apm-agent-core/src/main/java/co/elastic/apm/agent/report/ApmServerClient.java

```diff
@@ -346,6 +351,23 @@ Version getApmServerVersion(long timeout, TimeUnit timeUnit) throws InterruptedE
         return null;
     }
 
+    public boolean isLowerThan(Version apmServerVersion) {
+        if (this.apmServerVersion == null) {
+            throw new IllegalStateException("Called before init event");
+        }
+        try {
+            if (this.apmServerVersion.isDone()) {
```

Comment:
It's somewhat surprising that the semantics of `isLowerThan` and `isAtLeast` is somewhat different when it comes to blocking when the future isn't done yet.

Maybe just call `!isAtLeast(apmServerVersion)` from `isLowerThan` and pull up the `isDone` condition to supportsKeepingUnsampledTransaction. Add a comment explaining that this is to avoid blocking as the method is called in application threads.